skynet/dgpu/compute.py

```diff
@@ -81,153 +69,140 @@ def prepare_params_for_diffuse(
         _params
     )
 
+_model_name: str = ''
+_model_mode: str = ''
+_model = None
 
-# TODO, yet again - drop the redundant prefix ;)
-class SkynetMM:
-    '''
-    (AI algo) Model manager for loading models, computing outputs,
-    checking load state, and unloading when no-longer-needed/finished.
+@cm
+def maybe_load_model(name: str, mode: str):
+    if mode == 'diffuse':
+        mode = 'txt2img'
 
-    '''
-    def __init__(self, config: dict):
-        self.cache_dir = None
-        if 'hf_home' in config:
-            self.cache_dir = config['hf_home']
+    global _model_name, _model_mode, _model
+    config = load_skynet_toml().dgpu
 
-        self._model_name: str = ''
-        self._model_mode: str = ''
+    if _model_name != name or _model_mode != mode:
+        # unload model
+        _model = None
+        gc.collect()
+        torch.cuda.empty_cache()
 
-        # self.load_model(DEFAULT_INITAL_MODEL, 'txt2img')
+        _model_name = _model_mode = ''
 
-    def log_debug_info(self):
-        logging.info('memory summary:')
-        logging.info('\n' + torch.cuda.memory_summary())
+        # load model
```

Comment:
isn't this comment in the name 😉 

maybe a comment about what is "upscaling" and why we call a diff factor to create the `_model` in such cases?

Reply:
In `structify` branch i made `pipeline_for` function more generic and now it includes `init_upscaler` logic